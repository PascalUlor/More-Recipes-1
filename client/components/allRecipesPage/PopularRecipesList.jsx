import React from 'react';
import PropTypes from 'prop-types';
import PopularRecipe from './popularRecipesList/PopularRecipe.jsx';


const PopularRecipesList = ({ popularRecipes }) => {
  if (popularRecipes.length > 0) {
    return (
      <div className="card mb-5">
        <div className="card-body">
          <ul className="list-group list-group-flush mb-2">
            {popularRecipes
              .map((popularRecipe, index) => (index <= 3) &&
                <PopularRecipe
                  key={popularRecipe.id}
                  popularRecipe={popularRecipe} />)
            }
          </ul>
      </div>
    </div>
    );
  }
  return (
    <div></div>
  );
};

PopularRecipesList.propTypes = {
  popularRecipes: PropTypes.array.isRequired
};

export default PopularRecipesList;