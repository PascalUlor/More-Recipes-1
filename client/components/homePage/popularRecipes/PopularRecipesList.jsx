import React from 'react';
import PropTypes from 'prop-types';
import PopularRecipe from './PopularRecipe.jsx';


/**
 * @description displays a list of popular recipes
 * @method PopularRecipesList
 * 
 * @param { array } fetchedPopularRecipes - array of popular recipes
 * 
 * @returns { jsx } jsx - renders PopularRecipesList component
 */
const PopularRecipesList = ({ fetchedPopularRecipes }) => {
  const popularRecipes = fetchedPopularRecipes.map(popularRecipe =>
    <PopularRecipe key={popularRecipe.id} popularRecipe={popularRecipe} />);
  return (
    <div className="row">
      {popularRecipes}
    </div>
  );
};

PopularRecipesList.propTypes = {
  fetchedPopularRecipes: PropTypes.array.isRequired
};

export default PopularRecipesList;
