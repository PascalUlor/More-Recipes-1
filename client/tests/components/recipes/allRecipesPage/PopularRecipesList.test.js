/* eslint-disable */
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import PopularRecipesList
  from '../../../../components/allRecipesPage/PopularRecipesList.jsx';


const props = {
  popularRecipes: [{
    id: 1,
    title: '',
    User: { fullName: '' },
    upvotes: 1,
    downvotes: 2
  }, {
    id: 2,
    title: '',
    User: { fullName: '' },
    upvotes: 1,
    downvotes: 0
  }]
};

describe('AllRecipesList component', () => {
  it('should render correctly', () => {
    const mountWrapper = mount((
      <Router>
        <PopularRecipesList {...props}/>
      </Router>
    ));
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const newProps = {
      ...props,
      popularRecipes: []
    };
    const mountWrapper = mount((
      <Router>
        <PopularRecipesList {...newProps}/>
      </Router>
    ));
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });
});
