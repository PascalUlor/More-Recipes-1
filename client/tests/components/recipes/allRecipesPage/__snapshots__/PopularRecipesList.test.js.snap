// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllRecipesList component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PopularRecipesList
      popularRecipes={
        Array [
          Object {
            "User": Object {
              "fullName": "",
            },
            "downvotes": 2,
            "id": 1,
            "title": "",
            "upvotes": 1,
          },
          Object {
            "User": Object {
              "fullName": "",
            },
            "downvotes": 0,
            "id": 2,
            "title": "",
            "upvotes": 1,
          },
        ]
      }
    >
      <div
        className="card mb-5"
      >
        <div
          className="card-body"
        >
          <ul
            className="list-group list-group-flush mb-2"
          >
            <PopularRecipe
              key="1"
              popularRecipe={
                Object {
                  "User": Object {
                    "fullName": "",
                  },
                  "downvotes": 2,
                  "id": 1,
                  "title": "",
                  "upvotes": 1,
                }
              }
            >
              <li
                className="list-group-item"
              >
                <h5
                  className="text-muted text-center w-100 p-1"
                />
                <p
                  className="text-muted w-100 text-center"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user text-info"
                  />
                   
                </p>
                <div
                  className="text-center w-100"
                >
                  <small
                    className="text-success pr-1"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-thumbs-o-up"
                    />
                     
                    1
                  </small>
                  <Link
                    className="btn btn-outline-info btn-sm"
                    replace={false}
                    role="button"
                    to="recipes/1/recipe-details"
                  >
                    <a
                      className="btn btn-outline-info btn-sm"
                      href="recipes/1/recipe-details"
                      onClick={[Function]}
                      role="button"
                    >
                      <small>
                        View
                      </small>
                    </a>
                  </Link>
                  <small
                    className="text-danger pl-1"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-thumbs-o-down"
                    />
                     
                    2
                  </small>
                </div>
              </li>
            </PopularRecipe>
            <PopularRecipe
              key="2"
              popularRecipe={
                Object {
                  "User": Object {
                    "fullName": "",
                  },
                  "downvotes": 0,
                  "id": 2,
                  "title": "",
                  "upvotes": 1,
                }
              }
            >
              <li
                className="list-group-item"
              >
                <h5
                  className="text-muted text-center w-100 p-1"
                />
                <p
                  className="text-muted w-100 text-center"
                >
                  <i
                    aria-hidden="true"
                    className="fa fa-user text-info"
                  />
                   
                </p>
                <div
                  className="text-center w-100"
                >
                  <small
                    className="text-success pr-1"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-thumbs-o-up"
                    />
                     
                    1
                  </small>
                  <Link
                    className="btn btn-outline-info btn-sm"
                    replace={false}
                    role="button"
                    to="recipes/2/recipe-details"
                  >
                    <a
                      className="btn btn-outline-info btn-sm"
                      href="recipes/2/recipe-details"
                      onClick={[Function]}
                      role="button"
                    >
                      <small>
                        View
                      </small>
                    </a>
                  </Link>
                  <small
                    className="text-danger pl-1"
                  >
                    <i
                      aria-hidden="true"
                      className="fa fa-thumbs-o-down"
                    />
                     
                    0
                  </small>
                </div>
              </li>
            </PopularRecipe>
          </ul>
        </div>
      </div>
    </PopularRecipesList>
  </Router>
</BrowserRouter>
`;

exports[`AllRecipesList component should render correctly 2`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <PopularRecipesList
      popularRecipes={Array []}
    >
      <div />
    </PopularRecipesList>
  </Router>
</BrowserRouter>
`;
